README.md

```diff
@@ -143,3 +143,36 @@ Below is the complete list of available options that can be used to customize yo
 - **OVPN_K8S_DNS**: Kuberentes cluster dns server (required).
 - **OVPN_K8S_DH**: Openvpn dh.pem file path (default: /etc/openvpn/pki/dh.pem).
 - **OVPN_K8S_CERTS**: Openvpn certs.p12 file path (default: /etc/openvpn/pki/certs.p12).
+
+## Working on GKE
+
+Sensible values for the env vars when on Google Container Engine are:
+
+ - **OVPN_PROTO**:
+
+    This has to be TCP. `LoadBalancer`s only support TCP [as of kubernetes 1.0](http://kubernetes.io/v1.1/docs/user-guide/services.html).
+
+ - **OVPN_K8S_SERVICE_NETWORK**:
+
+    Get this by looking at the IPs of your running pods.
+
+        $kubectl get pods -o=template --template='{{range $index, $element := .items}}  {{$element.status.podIP}} {{$element.metadata.name}}
+        {{end}}'
+
+    This will show you the kind of range you need. eg for this set of pod ips:
+
+        10.84.2.37 podA
+        10.84.0.12 podB
+        10.84.1.4 podC
+
+    you would use 10.84.0.0
+
+ - **OVPN_K8S_SERVICE_SUBNET**:
+
+    This is based on the ip range wanted from above. In the example above,
+    use 255.255.0.0
+
+ - **OVPN_K8S_DNS**:
+ - **OVPN_NETWORK**:
+
+    Anything that doesn't clash with the OVPN_K8S_SERVICE_NETWORK.
```

Comment:
### Status for me

 - Using kubernetes via GKE
 - Env vars I'm using in my replication controller setup are:
```
- name: OVPN_NETWORK
  value: 10.240.0.0
- name: OVPN_SUBNET
  value: 255.255.0.0
- name: OVPN_PROTO
  value: tcp
- name: OVPN_K8S_SERVICE_NETWORK
  value: 10.84.0.0
- name: OVPN_K8S_SERVICE_SUBNET
  value: 255.255.0.0
- name: OVPN_K8S_DNS
  value: 10.241.240.10
```
 - I connect successfully to the openvpn running server [0]
```
Sat Jan  2 21:19:47 2016 [ploystvpnserver] Peer Connection Initiated with [AF_INET]104.155.91.74:1194
Sat Jan  2 21:19:50 2016 SENT CONTROL [ploystvpnserver]: 'PUSH_REQUEST' (status=1)
Sat Jan  2 21:19:50 2016 PUSH: Received control message: 'PUSH_REPLY,route 10.84.0.0 255.255.0.0,dhcp-option DOMAIN cluster.local,dhcp-option DNS 10.241.240.10,route 10.240.0.1,topology net30,ping 10,ping-restart 60,ifconfig 10.240.0.6 10.240.0.5'
Sat Jan  2 21:19:50 2016 OPTIONS IMPORT: timers and/or timeouts modified
Sat Jan  2 21:19:50 2016 OPTIONS IMPORT: --ifconfig/up options modified
Sat Jan  2 21:19:50 2016 OPTIONS IMPORT: route options modified
Sat Jan  2 21:19:50 2016 OPTIONS IMPORT: --ip-win32 and/or --dhcp-option options modified
Sat Jan  2 21:19:50 2016 Preserving previous TUN/TAP instance: utun0
Sat Jan  2 21:19:50 2016 Initialization Sequence Completed
```
 - Client machine is given a new network interface
```
utun0: flags=8051<UP,POINTOPOINT,RUNNING,MULTICAST> mtu 1500
    inet 10.240.0.6 --> 10.240.0.5 netmask 0xffffffff
    nd6 options=1<PERFORMNUD>
```
 - The OpenVPN server has following interfaces:
  - tun0 addr:10.240.0.1
  - eth0 addr:10.84.0.12
 - Pings from the client to both addresses timeout
```
➜  ~  ping 10.240.0.1
PING 10.240.0.1 (10.240.0.1): 56 data bytes
Request timeout for icmp_seq 0
Request timeout for icmp_seq 1
➜  ~  ping 10.84.0.12
PING 10.84.0.12 (10.84.0.12): 56 data bytes
Request timeout for icmp_seq 0
Request timeout for icmp_seq 1
```
 - I cannot telnet to `10.84.0.12 1194` nor `10.240.0.1 1194`. I can telnet via the external loadbalancer's address (clearly, as it is via this address that I connect to the server).
 - I also cannot telnet to any open ports on any other containers running on the cluster. 

### Thoughts

Initially I thought it was a port-forwarding thing but I see that you've got the `MASQUERADE` stuff set up in `entrypoint.sh`. Also, the fact that I'm not even able to hit the server pod raises questions for me.

Do you have/have you had these problems? 

[0] I occasionally get reset errors in the openvpn client:
```
Sat Jan  2 21:14:47 2016 Connection reset, restarting [-1]
Sat Jan  2 21:14:47 2016 SIGUSR1[soft,connection-reset] received, process restarting
```